web-src/src/components/GenerateButton.js

```diff
@@ -66,7 +66,7 @@ export function GenerateButton() {
 
   const generateResults = useCallback(async () => {
     const finalPrompt = renderPrompt(prompt, parameters);
-    const { queryId, response } = await firefallService.complete(finalPrompt, temperature, imsToken);
+    const { queryId, response } = await firefallService.complete(finalPrompt, temperature, user.imsToken);
```

Comment:
The IMS token should be injected into the service in a constructor. This way, we won't even touch this file.

Reply:
The EXC Shell triggers a refresh token when there is 20% duration left. A configuration event is sent to the iframe. The app could also listen directly to `user.on('change:imsToken', () => {})`

If we move this in a constructor we need to re-initialize `firefall` service when token changes.

Reply:
@askayastha22 Good point!

Reply:
Right now, the following shellEventsHandler recreates the application configuration on 'ready' and 'configuration' change.
```
runtime.on('ready', shellEventsHandler);
runtime.on('configuration', shellEventsHandler);
```

@ddragosd Do we really need to over-optimize this code?

Reply:
I just realized that any configuration change will recreate the FirefallService due to `shellEventsHandler` being called.